knowledge-base/customize-wpf-report-viewer.md

```diff
@@ -0,0 +1,143 @@
+---
+title: How to Customize the WPF Report Viewer
+description: Edit WPF Report Viewer template
+type: how-to
+page_title: Customize the WPF Report Viewer appearance
+slug: customize-wpf-report-viewer
+position: 
+tags: 
+ticketid: 1533694
+res_type: kb
+---
+
+## Environment
+<table>
+	<tbody>
+		<tr>
+			<td>Product</td>
+			<td>Progress® Telerik® Reporting</td>
+		</tr>
+		<tr>
+			<td>Report Viewer</td>
+			<td>WPF</td>
+		</tr>
+	</tbody>
+</table>
+
+
+## Description
+The article explains how to customize the appearance of the [WPF Report Viewer](../wpf-report-viewer-intro)
+
+## Suggested Workarounds
+Make sure you merge the resource dictionaries from the XAML files rather than from the default _Telerik.ReportViewer.Wpf.Themes_ assembly. See the article section [Setting Customizable Implicit Style for the WPF Report Viewer Example](../wpf-report-viewer-implicit-styles#setting-customizable-implicit-style-for-the-wpf-report-viewer-example) for details. The reason is that we need to modify the XAML files to add new controls and functionality to the viewer.
+
+Let's add two new buttons in the viewer's toolbar in _Telerik.ReportViewer.Wpf.xaml_ file of the corresponding theme:
+```XAML
+<ResourceDictionary
+    ...
+    xmlns:local="clr-namespace:CSharp.NetFramework.WpfIntegrationDemo">	<!-- The namespace for the custom RoutedUICommand -->
+    ...
+    <!-- Office2019 -->
+	  ...
+    <Style TargetType="telerikReporting:ReportViewer">
+        ...
+        <Setter Property="Template">
+            <Setter.Value>
+                <ControlTemplate TargetType="telerikReporting:ReportViewer">
+                    <Grid x:Name="LayoutRoot">
+                        <Border Background="{TemplateBinding Background}"
+                    BorderBrush="{TemplateBinding BorderBrush}"
+                    BorderThickness="{TemplateBinding BorderThickness}">
+                            <Grid Margin="{TemplateBinding Padding}">
+                                ...
+                                <!-- Toolbar -->
+                                <telerikNav:RadToolBar Grid.Row="0" Margin="0">
+                                    ...
+                                    <telerikNav:RadToolBarSeparator />
+                                    
+                                    <!-- the new Buttons in the ReportViewer's Toolbar -->
+                                    <Button x:Name="CustomA" Command="{Binding Path=MyCommand, RelativeSource={RelativeSource AncestorType=Window}}">Main Window Command</Button>
```

Comment:
Let's not make the purists angry - please name the buttons as advised by code naming guidelines: "customAButton" or "customCommandAButton".

Reply:
corrected